packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -669,12 +746,18 @@ export function makeVirtualObjectManager(
     vrm.registerKind(kindIDID, reanimateDurableKindID, () => null, true);
   }
 
-  function reanimateDurableKindID(vobjID, _proforma) {
+  function reanimateDurableKindID(vobjID) {
     const { subid: kindID } = parseVatSlot(vobjID);
     const raw = syscall.vatstoreGet(`vom.kind.${kindID}`);
     assert(raw, X`unknown kind ID ${kindID}`);
     const durableKindDescriptor = harden(JSON.parse(raw));
     const kindHandle = Far('kind', {});
+    // @ts-expect-error it would be great if TypeScript knew JavaScript
+    // eslint-disable-next-line no-proto
+    linkToCohort.set(kindHandle.__proto__, kindHandle);
```

Comment:
Maybe we should have a helper function `protect(obj, target)`, which walks the prototype chain, adding each prototype to `unweakable` and establishing the `linkToCohort`, until it reaches `Object.prototype`. I had a sudden fear that someone might change `Far` down the road to have *two* hoisted prototypes, and we'd never notice the security problem.

Reply:
Also it gets called like 4 times and it's currently 4-6 lines each, so it might be more concise too.

Reply:
I expect `FarClasses` will need something like this anyway.

Reply:
I started with something like that and it was a fail, because it turns out that things that need to make `unweakable` and things that need `linkToCohort` are not as correlated as one might imagine.